### PR TITLE
[IIO] restore iio device on _stop() @open sesame 4/2 15:17

### PR DESCRIPTION
**Pull request by kparichay**
restore the existing values of scan_channels_en, sampling_frequency,
buffer/length and current trigger (if exist) when stopping the element

**Related Issue**
#1199 

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_source/tensor_src_iio.c [hunk 1]
```diff
@@ -1317,6 +1332,18 @@ gst_tensor_src_iio_start (GstBaseSrc * src)
...
+    filename =
+        g_build_filename (self->device.base_dir, "sampling_frequency", NULL);
+    if (!g_file_get_contents (filename, &file_contents, &length, NULL)) {
+      GST_WARNING_OBJECT (self, "Unable to read default sampling frequency.");
+    } else if (file_contents != NULL) {
+      self->default_sampling_frequency =
+          g_ascii_strtoull (file_contents, NULL, 10);
+    }
+    g_free (file_contents);
+    g_free (filename);
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_source/tensor_src_iio.c]: This function is getting too long & complex. Can you refactor it later?

Comment (kparichay) [reply gst/nnstreamer/tensor_source/tensor_src_iio.c]: Yes, opened issue #1300

**Comment by myungjoo** [comment]
Need Rebase

**Comment by kparichay** [comment]
> Need Rebase

Done

**Pull request merged**